### PR TITLE
Some fix on Taxon API - Translation

### PR DESCRIPTION
**Pull request by AlucardleVash**
I add translation in taxon form.

Also i avoid to call getName in Taxon->__toString(), because a2lix create an empty entity for the root taxon, so it's not possible to add product link to taxon or taxon link to an other taxon.

Pierre

### CODE REVIEW

File: src/Sylius/Component/Taxonomy/Model/Taxon.php [hunk 1]
```diff
@@ -93,7 +93,7 @@ public function __construct()
      */
     public function __toString()
     {
-        return $this->translate()->__toString();
+        return is_null($this->getParent()) ? '---' : $this->translate()->__toString();
```
Comment (stloyd) [inline src/Sylius/Component/Taxonomy/Model/Taxon.php]: Should be `null === $this->getParent()` or at least `!$this->getParent()`.

Comment (AlucardleVash) [inline src/Sylius/Component/Taxonomy/Model/Taxon.php]: Yes, thanks

**Comment by pjedrzejewski** [comment]
@AlucardleVash Could you squash the commits? Thanks!

**Comment by AlucardleVash** [comment]
@pjedrzejewski it's done ;-)
Thank's @lbaey for git stash help :-)

**Comment by pjedrzejewski** [comment]
@AlucardleVash Sorry, but seems like specs are failing. Could you have a look? Let me know if you need any help!

File: src/Sylius/Component/Taxonomy/Model/Taxon.php [hunk 2]
```diff
@@ -93,7 +93,7 @@ public function __construct()
      */
     public function __toString()
     {
-        return $this->translate()->__toString();
+        return null === $this->getParent() ? '---' : $this->translate()->__toString();
```
Comment (aRn0D) [inline src/Sylius/Component/Taxonomy/Model/Taxon.php]: you should replace '---' by '-'

**Comment by aRn0D** [comment]
ping @AlucardleVash

**Comment by michalmarcinkowski** [comment]
@AlucardleVash could you rebase your PR?

**Pull request closed without merging**